sprokit/processes/core/detect_motion_process.cxx

```diff
@@ -0,0 +1,179 @@
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "detect_motion_process.h"
+
+#include <vital/util/wall_timer.h>
+#include <vital/algo/detect_motion.h>
+
+#include <sprokit/processes/kwiver_type_traits.h>
+#include <sprokit/pipeline/process_exception.h>
+
+namespace kwiver {
+
+create_config_trait( algo, std::string, "", "Algorithm configuration subblock" );
+
+//----------------------------------------------------------------
+// Private implementation class
+class detect_motion_process::priv
+{
+public:
+  priv();
+  ~priv();
+
+  vital::algo::detect_motion_sptr m_algo;
+  kwiver::vital::wall_timer m_timer;
+
+}; // end priv class
+
+
+// ==================================================================
+detect_motion_process::
+detect_motion_process( kwiver::vital::config_block_sptr const& config )
+  : process( config ),
+    d( new detect_motion_process::priv )
+{
+  make_ports();
+  make_config();
+}
+
+
+detect_motion_process::
+~detect_motion_process()
+{
+}
+
+
+// ------------------------------------------------------------------
+void
+detect_motion_process::
+_configure()
+{
+  scoped_configure_instrumentation();
+
+  vital::config_block_sptr algo_config = get_config();
+
+  // Check config so it will give run-time diagnostic of config problems
+  if ( ! vital::algo::detect_motion::check_nested_algo_configuration_using_trait( algo, algo_config ) )
+  {
+    throw sprokit::invalid_configuration_exception( name(), "Configuration check failed." );
+  }
+
+  vital::algo::detect_motion::set_nested_algo_configuration_using_trait( algo, algo_config, d->m_algo );
+
+  if ( ! d->m_algo )
+  {
+    throw sprokit::invalid_configuration_exception( name(), "Unable to create motion detector algorithm" );
+  }
+}
+
+
+// ------------------------------------------------------------------
+void
+detect_motion_process::
+_step()
+{
+  d->m_timer.start();
+
+  //TODO, handle case where this is optionally provided
+  //auto ts = grab_from_port_using_trait( timestamp );
```

Comment:
This comment appears to have been addressed.

Reply:
+1

Reply:
Comment deleted